src/selectedtests/work_items/process_task_mapping_work_items.py

```diff
@@ -62,11 +61,24 @@ def _process_one_task_mapping_work_item(
     """
     log = LOGGER.bind(project=work_item.project, module=work_item.module)
     log.info("Starting task mapping work item processing for work_item")
-    if _run_create_task_mappings(evg_api, mongo, work_item, after_date, log):
+    if _seed_task_mappings_for_project(evg_api, mongo, work_item, after_date, log):
         work_item.complete(mongo.task_mappings_queue())
 
 
-def _run_create_task_mappings(
+def _create_project_in_task_mappings_config(mongo, work_item, most_recent_version_analyzed):
+    mongo.task_mappings_project_config().insert_one(
```

Comment:
Kind of point in case: we know the database is mongodb at this point, and are using MongoClient's API with some interesting details (i.e. which collection?) being hidden from view.

IMO if you want to abstract mongo away it should look more like something along the lines of `data_service.save(domain_object)` rather than `data_proxy.insert_somewhere(db_representation)`.

Reply:
I refactored this so that ProjectConfig is a domain object now